ansicolor.js

```diff
@@ -196,133 +221,221 @@ const assignStringWrappingAPI = (target, wrapBefore = target) =>
 
 const TEXT    = 0,
       BRACKET = 1,
-      CODE    = 2
+      CODE    = 2;
+class Span {
+  constructor(code, text) {
+    this.code = code;
+    this.text = text;
+
+    // Those are added in the actual parse, this is done for performance reasons to have the same hidden class
+    this.css = "";
+    this.color = "";
+    this.bgColor = "";
+    this.bold = undefined;
+    this.inverse = undefined;
+    this.italic = undefined;
+    this.underline = undefined;
+    this.bright = undefined;
+    this.dim = undefined;
```

Comment:
Instead of adding those properties later only if the [`styles` have them](https://github.com/xpl/ansicolor/blob/5ce0946ad33ce08db760d741e7c00fac5f86ed0a/ansicolor.js#L295), we will always have those so we can take advantage of "hidden class"